status/status.go

```diff
@@ -148,60 +133,81 @@ OuterLoop:
 						resp.Body.Close()
 					}
 				}
-
-				// Mattermost
-				if configuration.MattermostServer != "" {
-					if err := UpdateMattermost(nc, status); err != nil {
-						log.Printf("Mattermost error: %s\n", err)
-					} else {
-						log.Printf("Mattermost updated")
-					}
-				}
 			}
 			first = false
 			time.Sleep(time.Second)
 		}
 	}
 }
 
-func UpdateMattermost(nc *http.Client, status bool) error {
-	mm := model.NewAPIv4Client(configuration.MattermostServer)
-	mm.HttpClient = nc
-	mm.SetToken(configuration.MattermostToken)
-
-	channel, resp := mm.GetChannel(configuration.MattermostChannelId, "")
-	if channel == nil {
-		return fmt.Errorf("Get channel: %+v", resp)
+// UpdateTopic modifies the topic (IRC, Mattermost) by matching `re` and replacing
+// the subexpression by `new`. The regexp must have exactly one subexpression.
```

Comment:
I feel like this should be enforced in a less error prone way, maybe error handle it better -- why check `if len(match) == 4` only when you could check for `== 0` too.

Reply:
Oh god i don't know how to use github

Reply:
Hmm like.. the cases are:

len(match) == 0: no match - someone broke the channel topic
len(match) == 1: golang std library bug lolwut?
len(match) == 2: 1 subexpr - coding bug - someone broken the contract on `re`
len(match) == 3: golang std library bug lolwut?
len(match) == 4: ahhh yesss crack open a cold one 🍺

I did not care enough to distinguish between the 0 .. 3 cases but I guess I could if you feel it's valuable?

Reply:
lol i mean idk i'm just finding things to point out - i was fine with it as ==4